### PR TITLE
Basic setup and single sanity test with sbr-control-api

### PR DESCRIPTION
**Pull request by ameenhaq**

### CODE REVIEW

File: src/test/resources/RequestGenerator.scala [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+/**
+  * RequestGenerator
+  * ----------------
+  * Author: haqa
+  * Date: 16 November 2017 - 09:25
+  * Copyright (c) 2017  Office for National Statistics
+  */
+
+@Singleton
+class RequestGenerator @Inject() (
```
Comment (cpcundill) [inline src/test/resources/RequestGenerator.scala]: Let's run this to see if Guice injection is working here.

**Review by cpcundill — DISMISSED** [review]
One comment so far.  Let's also add a Jenkinsfile and see this building before we merge.

File: Jenkinsfile [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+                    env.DEPLOY_NAME = DEPLOY_TEST
+                }
+                else if (BRANCH_NAME == BRANCH_PROD) {
+                    env.DEPLOY_NAME = DEPLOY_PROD
+                }
+                else {
+                    colourText("info", "No matching branch")
+                }
+                dir('conf') {
+                    git(url: "$GITLAB_URL/StatBusReg/${MODULE_NAME}.git", credentialsId: GITLAB_CREDS, branch: "${BRANCH_DEV}")
```
Comment (ONS-Tom) [inline Jenkinsfile]: should `${BRANCH_DEV}` be there?

File: Jenkinsfile.gdsl [hunk 3]
```diff
@@ -0,0 +1,95 @@
+
+//The global script scope
+def ctx = context(scope: scriptScope())
+contributor(ctx) {
```
Comment (ONS-Tom) [inline Jenkinsfile.gdsl:4]: Should this be in a `.gitignore`?

File: src/test/resources/reference.conf [hunk 4]
```diff
@@ -0,0 +1,60 @@
+ws {
+  request.timeout = 15000
+  request.timeout = ${?SBR_EXTERNAL_REQUEST_TIMEOUT}
+}
+
+data-item {
```
Comment (ONS-Tom) [inline src/test/resources/reference.conf:6]: does it make sense for all these to be in the `application.conf`?

**Review by ONS-Tom — APPROVED** [review]

File: src/test/scala/system/SystemSpec.scala [hunk 5]
```diff
@@ -14,17 +14,21 @@ class SystemSpec extends TestUtils {
...
+  private val SBR_CONTROL_API = "sbr-control-api"
+  private val SBR_ADMIN_DATA = "sbr-admin-data"
+
   private val defaultYearMonth = yearMonthConversion(defaultPeriod, REFERENCE_PERIOD_FORMAT)
 
 
   // TODO - fix children and parents testing null values
 
-  behavior of "sbr-api"
+  behavior of SBR_API
```
Comment (ONS-Tom) [inline src/test/scala/system/SystemSpec.scala]: should this be here?

File: src/test/scala/system/SystemSpec.scala [hunk 6]
```diff
@@ -200,13 +211,13 @@ class SystemSpec extends TestUtils {
 
 
 
-  behavior of "sbr-control"
+  behavior of SBR_CONTROL_API
```
Comment (ONS-Tom) [inline src/test/scala/system/SystemSpec.scala]: comment this out?

File: src/test/scala/system/SystemSpec.scala [hunk 7]
```diff
@@ -292,13 +309,13 @@ class SystemSpec extends TestUtils {
     */
 
 
-  behavior of "sbr-admin-data"
+  behavior of SBR_ADMIN_DATA
```
Comment (ONS-Tom) [inline src/test/scala/system/SystemSpec.scala]: comment this out?

File: src/test/scala/util/ApiTags.scala [hunk 8]
```diff
@@ -0,0 +1,28 @@
+package util
+
+import org.scalatest.Tag
+
+/**
+  * ApiTags
```
Comment (ONS-Tom) [inline src/test/scala/util/ApiTags.scala]: I like this approach

**Review by ONS-Tom — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by awharris — APPROVED** [review]

**Pull request merged**